src/cindent.c

```diff
@@ -2114,8 +2118,10 @@ get_c_indent(void)
 	goto laterend;
     }
 
-    // #defines and so on always go at the left when included in 'cinkeys'.
-    if (*theline == '#' && (*linecopy == '#' || in_cinkeys('#', ' ', TRUE)))
+    // #defines and so on go at the left when included in 'cinkeys',
+    // exluding pragmas when customized in 'cinoptions'
+    if (*theline == '#' && (*linecopy == '#' || in_cinkeys('#', ' ', TRUE))
+		&& (curbuf->b_ind_pragma == 0 || STRNCMP(theline, "#pragma", 7) != 0))
```

Comment:
This doesn't recognize `# pragma` (space between `#` and `pragma`).

Reply:
Fixed.